### PR TITLE
CSC Beam Halo filter: keep AOD info + bug fix

### PR DESCRIPTION
**Pull request by lathomas**
Bug fix in CSCHaloAlgo.cc: a boolean was not properly reinitialized in a loop. 
Also asking to keep the CSCHaloData class in AOD. (size: 128 bytes/62 per evt before/after compression)

### CODE REVIEW

File: RecoMET/Configuration/python/RecoMET_EventContent_cff.py [hunk 1]
```diff
@@ -64,6 +64,7 @@
                                            'drop recoHcalNoiseRBXs_*_*_*',
                                            'keep HcalNoiseSummary_hcalnoise_*_*',
                                            'keep *GlobalHaloData_*_*_*',
+                                           'keep *CSCHaloData_*_*_*',
```
Comment (slava77) [inline RecoMET/Configuration/python/RecoMET_EventContent_cff.py]: Please don't use wildcards on the first field.
(the neighboring lines better be updated to, but can be done by someone else in JetMET)

Comment (slava77) [inline RecoMET/Configuration/python/RecoMET_EventContent_cff.py]: Sorry, I meant the second field

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoMET/Configuration/python/RecoMET_EventContent_cff.py (hunk 1 above)
Comment (lathomas) [inline RecoMET/Configuration/python/RecoMET_EventContent_cff.py]: Done. I prefer not to touch other lines now but will try to find someone in JetMet that can have a look.

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
@lathomas 
please clarify the use case for this object:
- is it needed for MC event selections? 
- if it's needed for data selections: is it more important for 25ns data (2015C) or for 2015A/B (50 ns) ?

Clearly, the change will immediately apply only for new data taking
and a re-reco will be needed to get it into the previously collected data.

**Comment by mariadalfonso** [comment]
@slava77 
Would be important to have this fix in the 25ns data taking Prompt Reco (2015C).
If we can get it in the 2015B (4T, 50 ns) re-reco is a good bonus.

We do not strictly need this fix in the MC (beam halo is not simulated and the change in the overtagging will be minimal).
@lathomas @schoef

**Comment by cvuosalo** [comment]
+1

For #10652 1fe2df3a1acc31a5d2ab93ff162b4232bdeb2e5f

Very small bug fix for the beam halo filter used with CSC segments. Also, adding CSC halo data to AOD. There should be no changes to monitored quantities except those related to recovery of a few CSC segments and the added halo data.

The code changes are satisfactory, and Jenkins tests against baseline CMSSW_7_4_X_2015-08-09-1100 show no significant differences, except for the addition of a few CSC segments that the old buggy code missed, as shown below.  Also, the addition of the CSC halo data to AOD was verified for workflow 25202.0.

These plots are from workflow 4.53_RunPhoton2012B.

all_oldvsnew_runphoton2012bwf4p53c_recocschalodata_cschalodata__rereco_obj_nflathalosegments
all_oldvsnew_runphoton2012bwf4p53c_recobeamhalosummary_beamhalosummary__rereco_obj_cscloosehaloid

**Comment by davidlange6** [comment]
+1

**Pull request merged**